iocore/net/UnixNet.cc

```diff
@@ -454,31 +454,43 @@ NetHandler::mainNetEvent(int event, Event *e)
         vc->read.triggered = 1;
         if (get_ev_events(pd, x) & EVENTIO_ERROR) {
           vc->read.error = 1;
+          if (vc->read.enabled) {
+            if (!read_ready_list.in(vc)) {
+              read_ready_list.enqueue(vc);
+            }
+          }
+          if (vc->write.enabled) {
+            if (!write_ready_list.in(vc)) {
+              write_ready_list.enqueue(vc);
+            }
+          }
         } else {
           vc->read.error = 0;
-        }
-        if (!read_ready_list.in(vc)) {
-          read_ready_list.enqueue(vc);
-        } else if (get_ev_events(pd, x) & EVENTIO_ERROR) {
-          // check for unhandled epoll events that should be handled
-          Debug("iocore_net_main", "Unhandled epoll event on read: 0x%04x read.enabled=%d closed=%d read.netready_queue=%d",
-                get_ev_events(pd, x), vc->read.enabled, vc->closed, read_ready_list.in(vc));
+          if (!read_ready_list.in(vc)) {
+            read_ready_list.enqueue(vc);
+          }
         }
       }
       vc = epd->data.vc;
       if (get_ev_events(pd, x) & EVENTIO_WRITE) {
         vc->write.triggered = 1;
         if (get_ev_events(pd, x) & EVENTIO_ERROR) {
           vc->write.error = 1;
+          if (vc->read.enabled) {
+            if (!read_ready_list.in(vc)) {
+              read_ready_list.enqueue(vc);
+            }
+          }
+          if (vc->write.enabled) {
+            if (!write_ready_list.in(vc)) {
+              write_ready_list.enqueue(vc);
+            }
+          }
         } else {
           vc->write.error = 0;
-        }
-        if (!write_ready_list.in(vc)) {
-          write_ready_list.enqueue(vc);
-        } else if (get_ev_events(pd, x) & EVENTIO_ERROR) {
-          // check for unhandled epoll events that should be handled
-          Debug("iocore_net_main", "Unhandled epoll event on write: 0x%04x write.enabled=%d closed=%d write.netready_queue=%d",
-                get_ev_events(pd, x), vc->write.enabled, vc->closed, write_ready_list.in(vc));
+          if (!write_ready_list.in(vc)) {
+            write_ready_list.enqueue(vc);
+          }
```

Comment:
You just ignore the error if enabled==0. This is equal to revert #947 .
I think this is not a good fix to the issue.